### ISSUE TITLE
Overload doesn't work as expected when using ellipsis as a default argument value

### ISSUE THREAD

**Issue by saryou** [microsoft/pyright#1658]
**Describe the bug**
Overload doesn't work as expected when using ellipsis as a default argument value.

**To Reproduce**

```
from typing import Type, TypeVar, Generic, overload


class Base:
    pass


T = TypeVar('T')
BaseT = TypeVar('BaseT', bound=Base)


class Field(Generic[T]):
    pass


@overload
def field(type: Type[BaseT],
          *,
          default: BaseT = ...) -> Field[BaseT]:
    ...


@overload
def field(type: Type[int]) -> Field[int]:
    ...


def field(*args, **kwargs):
    raise NotImplementedError()


class MyClass(Base):
    pass


reveal_type(field(type=MyClass))
reveal_type(field(type=int))
```

mypy:

```
Revealed type is 'test.Field[test.MyClass*]'
Revealed type is 'test.Field[builtins.int]'
```

pyright:
```
Type of "field(type=MyClass)" is "Field[MyClass]"
Type of "field(type=int)" is "Field[...]"
```

**Expected behavior**
revealed type should be same as mypy's

**Additional context**

It is work as expected if I change `default: BaseT = ...` to `default: BaseT = None`

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

The problem is that the constraint solver was being told to assign `...` to type variable `BaseT`, and `...` was interpreted as an `Any` type. That caused the first overload to be selected inappropriately.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.124, which I just published. It will also be included in the next release of pylance.
